server/homebrew.api.js

```diff
@@ -17,344 +17,329 @@ const { DEFAULT_BREW, DEFAULT_BREW_LOAD } = require('./brewDefaults.js');
+					});
+			} else {
+				if(shouldDeleteGoogleBrew) {
+					// When there are still authors remaining, we delete the google brew but store the full brew in the Homebrewery database
+					brew.googleId = undefined;
+					brew.textBin = zlib.deflateRawSync(brew.text);
+					brew.text = undefined;
+				}
 
-	let brew = req.brew;
-	const { googleId, editId } = brew;
-	const account = req.account;
-	const isOwner = account && (brew.authors.length === 0 || brew.authors[0] === account.username);
-	// If the user is the owner and the file is saved to google, mark the google brew for deletion
-	const shouldDeleteGoogleBrew = googleId && isOwner;
-
-	if(brew._id) {
-		brew = _.assign(await HomebrewModel.findOne({ _id: brew._id }), brew);
-		if(account) {
-			// Remove current user as author
-			brew.authors = _.pull(brew.authors, account.username);
-			brew.markModified('authors');
+				// Otherwise, save the brew with updated author list
+				await brew.save()
+					.catch((err)=>{
+						throw { status: 500, message: err };
+					});
+			}
 		}
-
-		if(brew.authors.length === 0) {
-			// Delete brew if there are no authors left
-			await brew.remove()
+		if(shouldDeleteGoogleBrew) {
+			const deleted = await api.deleteGoogleBrew(account, googleId, editId, res)
 				.catch((err)=>{
 					console.error(err);
-					throw { status: 500, message: 'Error while removing' };
-				});
-		} else {
-			if(shouldDeleteGoogleBrew) {
-				// When there are still authors remaining, we delete the google brew but store the full brew in the Homebrewery database
-				brew.googleId = undefined;
-				brew.textBin = zlib.deflateRawSync(brew.text);
-				brew.text = undefined;
-			}
-
-			// Otherwise, save the brew with updated author list
-			await brew.save()
-				.catch((err)=>{
-					throw { status: 500, message: err };
+					res.status(500).send(err);
 				});
+			if(!deleted) return;
 		}
-	}
-	if(shouldDeleteGoogleBrew) {
-		const deleted = await deleteGoogleBrew(account, googleId, editId, res)
-			.catch((err)=>{
-				console.error(err);
-				res.status(500).send(err);
-			});
-		if(!deleted) return;
-	}
 
-	res.status(204).send();
+		res.status(204).send();
+	}
 };
 
-router.post('/api', asyncHandler(newBrew));
-router.put('/api/:id', asyncHandler(getBrew('edit', true)), asyncHandler(updateBrew));
-router.put('/api/update/:id', asyncHandler(getBrew('edit', true)), asyncHandler(updateBrew));
-router.delete('/api/:id', asyncHandler(deleteBrew));
-router.get('/api/remove/:id', asyncHandler(deleteBrew));
+router.post('/api', asyncHandler(api.newBrew));
+router.put('/api/:id', asyncHandler(api.getBrew('edit', true)), asyncHandler(api.updateBrew));
+router.put('/api/update/:id', asyncHandler(api.getBrew('edit', true)), asyncHandler(api.updateBrew));
+router.delete('/api/:id', asyncHandler(api.deleteBrew));
+router.get('/api/remove/:id', asyncHandler(api.deleteBrew));
 
-module.exports = {
-	homebrewApi : router,
-	getBrew
-};
+module.exports = api;
```

Comment:
By wrapping the whole API here, do we need to change the locations where `getbrew` was being used to now be API.getbrew? (I.e., in app.js)

Reply:
Nope! That's actually the beauty of it. Since the `module.exports` is set to `api`, nothing outside the file needs to change. As for `router`, it's set onto `api` by reference, so the routes still work when referencing it by `router` from within the file.

Reply:
Ah, you are right. It's just another object as before. Carry on :)